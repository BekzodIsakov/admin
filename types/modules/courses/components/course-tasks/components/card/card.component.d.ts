export function Card({ children, title, subtitle, link, active, norm, pending, button }: {
    children: any;
    title: any;
    subtitle: any;
    link: any;
    active: any;
    norm: any;
    pending: any;
    button: any;
}): JSX.Element;
//# sourceMappingURL=card.component.d.ts.map