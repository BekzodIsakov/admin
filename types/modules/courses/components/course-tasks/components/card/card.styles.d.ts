export const CardContainer: any;
export const CardContainerWithoutLink: any;
export const CardWithoutLink: any;
//# sourceMappingURL=card.styles.d.ts.map