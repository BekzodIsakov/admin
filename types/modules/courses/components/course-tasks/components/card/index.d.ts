export * from "./card.component";
//# sourceMappingURL=index.d.ts.map