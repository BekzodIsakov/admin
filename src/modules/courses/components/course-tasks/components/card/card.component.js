import React from 'react';
import { RoundContainer, Subtitle, Title } from '../atoms.style';
import {
  CardContainer,
  CardContainerWithoutLink,
  CardWithoutLink
} from './card.styles';

export const Card = ({
  children,
  title,
  subtitle,
  link,
  active,
  norm,
  pending,
  button
}) => {
  return link ? (
    <CardContainer
      target='_blank'
      download={link}
      href={link}
      subtitle={subtitle}
    >
      <RoundContainer
        norm={norm}
        active={active}
        pending={pending}
        justify='center'
        align='center'
      >
        {children}
      </RoundContainer>
      <div>
        {title && <Title data-testid='course-task-linkTitle'>{title}</Title>}
        {subtitle && (
          <Subtitle data-testid='course-task-linkSubtitle'>{subtitle}</Subtitle>
        )}
      </div>
    </CardContainer>
  ) : (
    <CardWithoutLink>
      <CardContainerWithoutLink subtitle={subtitle}>
        <RoundContainer
          norm={norm}
          active={active}
          pending={pending}
          justify='center'
          align='center'
        >
          {children}
        </RoundContainer>
        <div>
          {title && <Title data-testid='course-task-item-title'>{title}</Title>}
          {subtitle && (
            <Subtitle data-testid='course-task-item-subtitle'>
              {subtitle}
            </Subtitle>
          )}
        </div>
      </CardContainerWithoutLink>
      {button}
    </CardWithoutLink>
  );
};
