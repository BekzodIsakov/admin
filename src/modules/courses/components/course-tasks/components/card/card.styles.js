import styled from 'styled-components';

export const CardContainer = styled.a`
  display: flex;
  gap: 1.6rem;
  min-width: 32.8rem;
  padding: 1.6rem;
  border-radius: 0.4rem;
  box-shadow: 0 0.4rem 1.5rem rgba(0, 0, 0, 0.05);
  text-decoration: none;
  color: #000000;

  & > div:last-child {
    display: flex;
    flex-direction: column;
    justify-content: ${props => (props.subtitle ? 'space-between' : 'center')};
    height: 4.4rem;
  }
`;

export const CardContainerWithoutLink = styled.div`
  display: flex;
  gap: 1.6rem;
  min-width: 32.8rem;
  text-decoration: none;
  color: #000000;

  & > div:last-child {
    display: flex;
    flex-direction: column;
    justify-content: ${props => (props.subtitle ? 'space-between' : 'center')};
    height: 4.4rem;
  }
`;

export const CardWithoutLink = styled.div`
  display: flex;
  justify-content: space-between;
  box-shadow: 0 0.4rem 1.5rem rgba(0, 0, 0, 0.05);
  border-radius: 0.4rem;
  padding: 1.6rem;
  align-items: center;
`;
