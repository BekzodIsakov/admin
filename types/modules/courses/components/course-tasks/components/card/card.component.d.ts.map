{"version":3,"file":"card.component.d.ts","sourceRoot":"","sources":["../../../../../../../src/modules/courses/components/course-tasks/components/card/card.component.js"],"names":[],"mappings":"AAQO;;;;;;;;;gBAmDN"}