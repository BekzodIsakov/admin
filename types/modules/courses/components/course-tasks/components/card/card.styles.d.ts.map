{"version":3,"file":"card.styles.d.ts","sourceRoot":"","sources":["../../../../../../../src/modules/courses/components/course-tasks/components/card/card.styles.js"],"names":[],"mappings":"AAEA,gCAgBE;AAEF,2CAaE;AAEF,kCAOE"}