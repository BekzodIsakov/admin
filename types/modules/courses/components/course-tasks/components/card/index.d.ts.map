{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../../../../src/modules/courses/components/course-tasks/components/card/index.js"],"names":[],"mappings":""}