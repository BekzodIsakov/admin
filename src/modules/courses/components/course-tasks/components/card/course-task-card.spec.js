import { customRender } from '@test-utils/helpers';
import { screen } from '@testing-library/react';

import { Card } from './card.component';

const setup = ({
  children,
  title,
  subtitle,
  link,
  active,
  norm,
  pending,
  button
}) => {
  return customRender(
    <Card
      children={children}
      title={title}
      subtitle={subtitle}
      link={link}
      active={active}
      norm={norm}
      pending={pending}
      button={button}
    />
  );
};

test('Course task card', () => {
  const { asFragment } = setup({
    children: 'children',
    title: 'title',
    subtitle: 'subtitle',
    link: false,
    active: false,
    norm: false,
    pending: false,
    button: ''
  });
  expect(asFragment).toMatchSnapshot();
});

test('course task card check props', () => {
  setup({
    children: 'children',
    title: 'title',
    subtitle: 'subtitle',
    link: false,
    active: false,
    norm: false,
    pending: false,
    button: ''
  });
  const courseTaskCardTitle = screen.getByTestId('course-task-item-title');
  const courseTaskCardText = screen.getByTestId('course-task-item-subtitle');
  expect(courseTaskCardTitle && courseTaskCardText).toBeTruthy();
});

test('course task card check props with link', () => {
  setup({
    children: 'children',
    title: 'title',
    subtitle: 'subtitle',
    link: '/',
    active: false,
    norm: false,
    pending: false,
    button: ''
  });
  const courseTaskCardTitle = screen.getByTestId('course-task-linkTitle');
  const courseTaskCardText = screen.getByTestId('course-task-linkSubtitle');
  expect(courseTaskCardTitle && courseTaskCardText).toBeTruthy();
});
