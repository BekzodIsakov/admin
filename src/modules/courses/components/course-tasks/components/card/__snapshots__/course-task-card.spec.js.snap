// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course task card 1`] = `[Function]`;
